packages/mdc-line-ripple/index.js

```diff
@@ -17,44 +17,61 @@
 
 import MDCComponent from '@material/base/component';
 
-import MDCTextFieldBottomLineAdapter from './adapter';
-import MDCTextFieldBottomLineFoundation from './foundation';
+import MDCLineRippleAdapter from './adapter';
+import MDCLineRippleFoundation from './foundation';
 
 /**
- * @extends {MDCComponent<!MDCTextFieldBottomLineFoundation>}
+ * @extends {MDCComponent<!MDCLineRippleFoundation>}
  * @final
  */
-class MDCTextFieldBottomLine extends MDCComponent {
+class MDCLineRipple extends MDCComponent {
   /**
    * @param {!Element} root
-   * @return {!MDCTextFieldBottomLine}
+   * @return {!MDCLineRipple}
    */
   static attachTo(root) {
-    return new MDCTextFieldBottomLine(root);
+    return new MDCLineRipple(root);
   }
 
   /**
-   * @return {!MDCTextFieldBottomLineFoundation}
+   * Activates the line ripple
    */
-  get foundation() {
-    return this.foundation_;
+  activate() {
+    this.foundation_.activate();
   }
 
   /**
-   * @return {!MDCTextFieldBottomLineFoundation}
+   * Deactivates the line ripple
+   */
+  deactivate() {
+    this.foundation_.deactivate();
+  }
+
+  /**
+   * Sets the transform origin given a user's click location. The `rippleCenter` is the
+   * x-coordinate of the middle of the ripple.
+   * @param {!number} rippleCenter
+   */
+  setTransformOrigin(rippleCenter) {
+    this.foundation_.setTransformOrigin(rippleCenter);
+  }
+
+  /**
+   * @return {!MDCLineRippleFoundation}
    */
   getDefaultFoundation() {
-    return new MDCTextFieldBottomLineFoundation(/** @type {!MDCTextFieldBottomLineAdapter} */ (Object.assign({
+    return new MDCLineRippleFoundation(/** @type {!MDCLineRippleAdapter} */ (Object.assign({
       addClass: (className) => this.root_.classList.add(className),
       removeClass: (className) => this.root_.classList.remove(className),
+      hasClass: (className) => this.root_.classList.contains(className),
       setAttr: (attr, value) => this.root_.setAttribute(attr, value),
       registerEventHandler: (evtType, handler) => this.root_.addEventListener(evtType, handler),
       deregisterEventHandler: (evtType, handler) => this.root_.removeEventListener(evtType, handler),
       notifyAnimationEnd: () => {
-        this.emit(MDCTextFieldBottomLineFoundation.strings.ANIMATION_END_EVENT, {});
+        this.emit(MDCLineRippleFoundation.strings.ANIMATION_END_EVENT, {});
```

Comment:
I thought you were going to remove this ANIMATION_END_EVENT? It doesn't seem necessary.